### PR TITLE
G5K job id support

### PR DESCRIPTION
**Pull request by Kyominii**
* Let users to give G5K job id into reservation.yaml and inform its type (oargrid or regular oar job) => https://github.com/BeyondTheClouds/enos/issues/251

### CODE REVIEW

File: enos/provider/g5k.py [hunk 1]
```diff
@@ -54,8 +54,16 @@ def init(self, conf, force_deploy=False):
         return (roles, network, (network_interface, external_interface))
 
     def destroy(self, env):
+
+        gridjob = None
         provider_conf = env['config']['provider']
-        gridjob, _ = EX5.planning.get_job_by_name(provider_conf['name'])
+
+        if 'jobid' in provider_conf:
```
Comment (rcherrueau) [inline enos/provider/g5k.py]: May I suggest to keep it in one line, and thus remove unnecessary `if` control flow e.g.,
```python
gridjob = provider_conf.get('jobid', EX5.planning.get_job_by_name(provider_conf['name'][0]))
```

**Comment by rcherrueau** [comment]
I did not notice the second commit. Do not take into account my first review, I have to look at the second commit first.

File: enos/provider/g5k.py (hunk 1 above)
Comment (Kyominii) [reply enos/provider/g5k.py]: I've just modified these `if` in the second commit

**Comment by msimonin** [comment]
So the goal is to support a job_id to be specified in the configuration.
- Supporting a oargrid jobid is adressed in the first commit
- Supporting an oar or an oargrid jobid is addressed by the second commit

The difficult part is the second one: supporting both oar and oargrid. 

To this ends a variable `oargrid: True|False` serve to instruct that the `jobid` passed is actually a `oargrid` job id or no.
I think this key in the configuration is confusing because if `oargrid: False` and no `jobid` is set in the reservation, one may think this will reserve the resource using oar instead of oargrid. This seems to not be the case.

One quick idea could be to have two possible new keys in the configuration:
- `oargrid_jobid`
- `oar_jobid` (and `frontend`)

With the following behaviour:
- if None of them is specified, reservation is made using oargrid
- if exactly one of them is specified, act accordingly (reload/delete from oargrid or oar)
- if both are specified a priority is given to one of them or an error is thrown.

In the future we may want to support this in Enoslib where we could enforce some validation.

ideas ?

**Comment by rcherrueau** [comment]
I have read the commit, but have to prepare the Stack seminar for tomorrow. So I will give my thought on this in due time. Thanks @msimonin for the sum up.

**Comment by Kyominii** [comment]
I modified the second commit to take into account your comments.

**Comment by msimonin** [comment]
This has been implemented in enoslib

**Pull request closed without merging**